data/wb_presets.json

```diff
@@ -25942,6 +25942,1154 @@
+                2.291015625,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": -2,
+              "channels": [
+                1.2998046875,
+                1,
+                2.255859375,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": -1,
+              "channels": [
+                1.3193359375,
+                1,
+                2.216796875,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "channels": [
+                1.33984375,
+                1,
+                2.1787109375,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 1,
+              "channels": [
+                1.3603515625,
+                1,
+                2.146484375,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 2,
+              "channels": [
+                1.3779296875,
+                1,
+                2.111328125,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 3,
+              "channels": [
+                1.400390625,
+                1,
+                2.0771484375,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 4,
+              "channels": [
+                1.421875,
+                1,
+                2.0439453125,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 5,
+              "channels": [
+                1.4462890625,
+                1,
+                2.01171875,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 6,
+              "channels": [
+                1.46875,
+                1,
+                1.98046875,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 7,
+              "channels": [
+                1.490234375,
+                1,
+                1.9462890625,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 8,
+              "channels": [
+                1.5146484375,
+                1,
+                1.91796875,
+                0
+              ]
+            },
+            {
+              "name": "Tungsten",
+              "tuning": 9,
+              "channels": [
+                1.5400390625,
+                1,
+                1.8857421875,
+                0
+              ]
+            },
+            {
+              "name": "CoolWhiteFluorescent",
```

Comment:
All of these and similar need to be space separated, e.g. "Cool White Fluorescent"...

Reply:
Sorry, not sure I follow? These are two different/unrelated settings... All of them need to be space separated.

Reply:
The point is that the same preset in camera and in exif (called "White fluorescent light" in camera menu and "Fluorescent" in exif) is translated by the `tools/extract_wb.py` script as "WhiteFluorescent" for untuned (no amber/green shift) and as "CoolWhiteFluorescent" for tuned presets (with amber/green shift). It's likely a mistake or an omission but I'm not sure which name is prefered. Note that existing presets for Canon R5 show exactly the same inconsistency (not for R6 as it has only untuned presets provided).

Reply:
Ah, can't really comment on that without digging into the actual raw files (or text output generated by `tools/extract_wb_from_images.sh`). Feel free to poke around and share any findings or more details.

Reply:
You can also double check if there are any new Canon specific tags from https://exiftool.org/TagNames/Canon.html we need to add/handle in both scripts.

Reply:
Ok, I see now for many EOS R series cameras there is a duplicated "White Fluorescent" entry, no idea where it came from... It seems there is already a correct untuned (silent "tuning": 0) entry present in the "Cool White Fluorescent" sequence, no?

Reply:
Well, THAT workaround worked for canon's DSLRs AFAIK and was there before I translated that script from Ruby into (in theory) more maintainable Python :) so yeah, we need somebody with raw samples for all WB finetuning to be able to fix this.

Reply:
Btw, the set from Canon EOS R7 here seems complete: https://github.com/darktable-org/darktable/issues/13153 (6 x 19 presets)

Reply:
It seems so, at least it's the same with R6 Mark II: camera menu says "White fluorescent light" and the table in manual lists this setting as 4000K.

Reply:
Updated the branch: added spaces to preset names and omitted the duplicate entry for untuned "Cool White Fluorescent" preset.